apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v1/x/log/GRPCLogAppenderInterceptor.java

```diff
@@ -92,12 +93,16 @@ private LogData transform(LoggingEvent event) {
                                 .setKey("thread").setValue(event.getThreadName()).build())
                         .build())
                 .setBody(LogDataBody.newBuilder().setType(LogDataBody.ContentCase.TEXT.name())
-                        .setText(TextLog.newBuilder().setText(event.getMessage()).build()).build());
+                                    .setText(TextLog.newBuilder().setText(transformLogText(event)).build()).build());
         return -1 == ContextManager.getSpanId() ? builder.build()
                 : builder.setTraceContext(TraceContext.newBuilder()
                         .setTraceId(ContextManager.getGlobalTraceId())
                         .setSpanId(ContextManager.getSpanId())
                         .setTraceSegmentId(ContextManager.getSegmentId())
                         .build()).build();
     }
+
+    private String transformLogText(final LoggingEvent event) {
+        return event.getMessage() + "\n" + ThrowableTransformer.INSTANCE.convert2String(event.getThrowable(), 2048);
```

Comment:
I'm wondering whether this `2048` deserves a new config item, or the default value is suitable in this case @wu-sheng .

Reply:
Let's keep this as default only. After all, in the next 8.5.0, I am expecting to use the file scraper to read the logs.